linux/disk.go

```diff
@@ -376,6 +489,40 @@ func addPartitionSet(d disko.Disk, pSet disko.PartitionSet) error { //nolint: fu
 		return err
 	}
 
+	return nil
+}
+
+// addPartitionSet - open the disk, add partitions.
+//     Caller's responsibility to udevSettle
+func addPartitionSet(d disko.Disk, pSet disko.PartitionSet) error {
+	if d.Table != disko.MBR && d.Table != disko.GPT && d.Table != disko.TableNone {
+		return fmt.Errorf("cannot add partition disk %s with table type %s", d.Name, d.Table)
+	}
+
+	if err := rangeCheckParts(d, pSet); err != nil {
+		return err
+	}
+
+	fp, err := os.OpenFile(d.Path, os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	defer fp.Close()
+
+	if err := syscall.Flock(int(fp.Fd()), unix.LOCK_EX); err != nil {
+		return fmt.Errorf("failed to lock %s: %s", d.Path, err)
+	}
+
+	if d.Table == disko.MBR {
+		err = addPartitionSetMBR(fp, d, pSet)
```

Comment:
One thing you could do here is do something like:
```
type PartitionType interface {
    AddPartitionSet(...)
    ...
}

type MBR struct {}
func (p MBR) AddPartitionSet(...) { }

type GPT struct {}
func (p GPT) AddPartitionSet(...) {}
```

which would let you get away from sprinkling the code with "if d.Table == ..." everywhere. Maybe useful maybe not, but I guess the more functions you add per partition table the nicer it might be.

Reply:
@tych0,
yeah, I realize i could have done that.  I think at the moment its not too bad.

Thank you for the suggestion and the review.